warehouse/tuf/services.py

```diff
@@ -0,0 +1,407 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import datetime
+import glob
+import os.path
+import shutil
+import warnings
+
+from contextlib import contextmanager
+
+from securesystemslib.exceptions import StorageError
+from securesystemslib.interface import (
+    import_ed25519_privatekey_from_file,
+    import_ed25519_publickey_from_file,
+)
+from zope.interface import implementer
+
+from warehouse.config import Environment
+from warehouse.tuf.constants import BIN_N_COUNT, Role
+from warehouse.tuf.hash_bins import HashBins
+from warehouse.tuf.interfaces import IKeyService, IRepositoryService, IStorageService
+from warehouse.tuf.repository import (
+    TOP_LEVEL_ROLE_NAMES,
+    MetadataRepository,
+    RolesPayload,
+    TargetsPayload,
+)
+
+
+class InsecureKeyWarning(UserWarning):
+    pass
+
+
+@implementer(IKeyService)
+class LocalKeyService:
+    """
+    A service to read public and private TUF role keys as local files for development.
+
+    NOTE: Do not use in production!
+    """
+
+    def __init__(self, key_path, request):
+        warnings.warn(
+            "LocalKeyService is intended only for use in development, you "
+            "should not use it in production to avoid unnecessary key exposure.",
+            InsecureKeyWarning,
+        )
+
+        self._key_path = key_path
+        self._request = request
+
+    @classmethod
+    def create_service(cls, context, request):
+        return cls(request.registry.settings["tuf.key.path"], request)
+
+    def get(self, rolename, key_type):
+        """
+        Returns Key objects for passed TUF role name from configured TUF key path. Key
+        type is one of 'public' or 'private'.
+        """
+        if key_type == "private":
+            privkey_path = os.path.join(self._key_path, f"{rolename}*")
+            role_keys = glob.glob(privkey_path)
```

Comment:
The order doesn't matter because we don't use the "public" key, as we always use the "private" that has the portion of the public key.
I added it separated, but I guess we could remove it. Any concern?